File: tf2-item-scanner/src/com/minowak/scanner/engine/SteamUser.java
package com.minowak.scanner.engine;



import java.io.IOException;
import java.net.MalformedURLException;
import java.util.LinkedList;
import java.util.List;

import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;

import com.minowak.scanner.gui.MainWindow;
import com.minowak.scanner.schema.ItemQuality;
import com.minowak.scanner.utils.Configuration;

public class SteamUser extends SteamEntity {
	private String apiUrl;
	private String api2Url;
	private String api3Url;
	private String tf2opUrl;
	private String id;
	private Backpack backpack;
	private long timePlayed;
	private long online;
	private List<String> friends = new LinkedList<String>();
	private String name;
	private boolean hasOP;

	public SteamUser(String id) {
		this.id = id;
		this.apiUrl = String.format("http://api.steampowered.com/IPlayerService/GetRecentlyPlayedGames/v0001/?key=%s&format=json&SteamID=%s",
				Configuration.API_KEY, id);
		this.api2Url = String.format("http://api.steampowered.com/ISteamUser/GetPlayerSummaries/v2/?key=%s&steamids=%s",
				Configuration.API_KEY, id);
		this.api3Url = String.format("http://api.steampowered.com/ISteamUser/GetFriendList/v0001/?key=%s&steamid=%s&relationship=friend",
				Configuration.API_KEY, id);
		this.tf2opUrl = "http://www.tf2outpost.com/user/";
	}

	public boolean isPremium() {
		return backpack.getSize() >= 300;
	}

	public long played() {
		return timePlayed;
	}

	public long lastOnline() {
		return online;
	}

	public String getName() {
		return name;
	}

	public boolean init() throws ParseException {
		JSONParser parser = new JSONParser();
		String jsonResponse = super.getJson(apiUrl);
		if(jsonResponse == null) {
			MainWindow.LOGGER.info("JSON response is null");
			return false;
		}
		Object responseObj = parser.parse(jsonResponse);
		JSONObject response = (JSONObject)((JSONObject) responseObj).get("response");
		JSONArray games = (JSONArray)response.get("games");


		if(games != null) {
			for(int i = 0 ; i < games.size() ; i++) {
				JSONObject game = (JSONObject)games.get(i);
				if(game.get("appid").equals("440")) {
					timePlayed = (long) game.get("playtime_forever");
					break;
				}
			}
		}

		jsonResponse = super.getJson(api2Url);
		if(jsonResponse == null) {
			MainWindow.LOGGER.info("JSON response is null");
			return false;
		}
		responseObj = parser.parse(jsonResponse);
		response = (JSONObject)((JSONObject) responseObj).get("response");
		JSONObject player = (JSONObject)((JSONArray) response.get("players")).get(0);
		online = (long)player.get("lastlogoff");
		name = (String)player.get("personaname");

		jsonResponse = super.getJson(api3Url);
		if(jsonResponse == null) {
			MainWindow.LOGGER.info("JSON response is null. API KEY=" + Configuration.API_KEY);
			return false;
		}
		responseObj = parser.parse(jsonResponse);
		response = (JSONObject)((JSONObject) responseObj).get("friendslist");
		JSONArray friendsArray = (JSONArray) response.get("friends");

		for(int i = 0 ; i < friendsArray.size() ; i++) {
			JSONObject friend = (JSONObject)friendsArray.get(i);
			friends.add((String)friend.get("steamid"));
		}

		try {
			if(getResponseCode(tf2opUrl + id) == 404) {
				hasOP = false;
			} else {
				hasOP = true;
			}
		} catch (IOException e) {
			hasOP = false;
			MainWindow.LOGGER.info("Error while checking tf2op. Assuming he dont have it");
		}

		backpack = new Backpack(id);
		return backpack.init();
	}

	public List<String> getFriendsIds() {
		return friends;
	}

	public boolean hasOutpost() {
		return hasOP;
	}

	public boolean hasItem(long itemId, ItemQuality quality) {
		return backpack.hasItem(itemId, quality);
	}

	public boolean hasUnusual() {
		return backpack.hasUnusual();
	}

	public double getValue() {
		try {
			return Backpack.getValue(id);
		} catch (ParseException e) {
			return 0.0;
		}
	}
}

File: tf2-item-scanner/tf2-item-scanner/engine/SteamUser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using tf2_item_scanner.utils;
using Newtonsoft.Json.Linq;

namespace tf2_item_scanner.engine
{
    class SteamUser
    {
        private string _apiUrl;
        private string _apiUrl2;
        private string _apiUrl3;
        private string _id;
        private Backpack _backpack;
        private long _timePlayed;
        private long _lastOnline;
        private List<string> _friends = new List<string>();
        private string _name;
        private string _state;

        public SteamUser(string id)
        {
            _id = id;
            _apiUrl = Utils.OwnedGamesUrl + id;
            _apiUrl2 = Utils.SummariesUrl + id;
            _apiUrl3 = Utils.FriendsUrl + id;
        }

        #region PROPERTIES

        public string Id
        {
            get { return _id; }
        }

        public long PlayTime
        {
            get { return _timePlayed; }
        }

        public long LastOnline
        {
            get { return _lastOnline; }
        }

        public string Name
        {
            get { return _name; }
        }

        #endregion

        public bool IsPremium()
        {
            return _backpack.Size >= 300;
        }

        public bool Init()
        {
            try
            {
                string json = Utils.GetJson(_apiUrl);
                JObject o = JObject.Parse(json);

                JObject response = (JObject)o["response"];

                JArray games = (JArray)response["games"];

                if (games != null)
                {
                    for (int i = 0; i < games.Count; i++)
                    {
                        JObject game = (JObject)games[i];
                        if ((string)game["appid"] == "440")
                        {
                            _timePlayed = (long)game["playtime_forever"];
                            break;
                        }
                    }
                }

                json = Utils.GetJson(_apiUrl2);
                o = JObject.Parse(json);

                response = (JObject)o["response"];
                JObject player = (JObject)((JArray)response["players"])[0];
                _lastOnline = (long)player["lastlogoff"];
                _name = (string)player["personaname"];
                switch ((int)player["personastate"])
                {
                    case 0: _state = "Offline"; break;
                    case 5:
                    case 6:
                    case 1: _state = "Online"; break;
                    case 2: _state = "Busy"; break;
                    case 3: _state = "Away"; break;
                    case 4: _state = "Snooze"; break;
                }

                json = Utils.GetJson(_apiUrl3);

                o = JObject.Parse(json);
                response = (JObject)o["friendslist"];
                JArray friendsArray = (JArray)response["friends"];

                for (int i = 0; i < friendsArray.Count; i++)
                {
                    JObject friend = (JObject)friendsArray[i];
                    _friends.Add((string)friend["steamid"]);
                }

                _backpack = new Backpack(_id);

                return _backpack.Init();
            }
            catch (Exception e)
            {
                return false;
            }
        }

        #region PROPERTIES

        public List<string> FriendsIds
        {
            get { return _friends; }
        }

        public string State
        {
            get { return _state; }
        }

        public double Value
        {
            get { return _backpack.Value; }
        }

        public long ItemsCount
        {
            get { return _backpack.ItemsCount; }
        }

        #endregion

        public bool HasItem(TF2Item item)
        {
            return _backpack.HasItem(item);
        }

        public TF2Item HasUnusual()
        {
            return _backpack.HasUnusual();
        }
    }
}
